arm_characterization/data_analyzer.py

```diff
@@ -33,7 +33,7 @@ def isfloat(value):
         return False
 
 
-mainGUI = tkinter.Tk()
+mainGUI = None
```

Comment:
Why the global?

Reply:
Sorry about the slow reply. This was global in the original program, so I kept it that way. This variable is used throughout, which is probably why it was setup in the way it is.

It might be cleaner to change `mainGUI` so that it's passed around instead. If you would prefer that please let me know and I'll deal with it.

Reply:
I guess I never looked at the other one. I suppose this is fine for now.